drivers/ad5770r/ad5770r.c

```diff
@@ -0,0 +1,718 @@
+ */
+int32_t ad5770r_set_mask_channel(struct ad5770r_dev *dev,
+				 const struct ad5770r_channel_switches *mask_channel_sel)
+{
+	int32_t ret;
+
+	if (!dev || !mask_channel_sel)
+		return FAILURE;
+
+	ret = ad5770r_spi_reg_write(dev,
+				    AD5770R_CH_SELECT,
+				    AD5770R_CH_SELECT_SEL_CH(mask_channel_sel->en0, AD5770R_CH0) |
+				    AD5770R_CH_SELECT_SEL_CH(mask_channel_sel->en1, AD5770R_CH1) |
+				    AD5770R_CH_SELECT_SEL_CH(mask_channel_sel->en2, AD5770R_CH2) |
+				    AD5770R_CH_SELECT_SEL_CH(mask_channel_sel->en3, AD5770R_CH3) |
+				    AD5770R_CH_SELECT_SEL_CH(mask_channel_sel->en4, AD5770R_CH4) |
+				    AD5770R_CH_SELECT_SEL_CH(mask_channel_sel->en5, AD5770R_CH5));
+
+	if (ret)
+		return ret;
+
+	dev->mask_channel_sel = *mask_channel_sel;
+
+	return ret;
+}
+
+/**
+ * Set software LDAC.
+ * @param dev - The device structure.
+ * @param sw_ldac - The array contains channel selection options.
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+int32_t ad5770r_set_sw_ldac(struct ad5770r_dev *dev,
+			    const struct ad5770r_channel_switches *sw_ldac)
+{
+	int32_t ret;
+
+	if (!dev || !sw_ldac)
+		return FAILURE;
+
+	ret = ad5770r_spi_reg_write(dev,
+				    AD5770R_SW_LDAC,
+				    AD5770R_SW_LDAC_CH(sw_ldac->en0, AD5770R_CH0) |
+				    AD5770R_SW_LDAC_CH(sw_ldac->en1, AD5770R_CH1) |
+				    AD5770R_SW_LDAC_CH(sw_ldac->en2, AD5770R_CH2) |
+				    AD5770R_SW_LDAC_CH(sw_ldac->en3, AD5770R_CH3) |
+				    AD5770R_SW_LDAC_CH(sw_ldac->en4, AD5770R_CH4) |
+				    AD5770R_SW_LDAC_CH(sw_ldac->en5, AD5770R_CH5));
+
+	if (ret)
+		return ret;
+
+	dev->sw_ldac = *sw_ldac;
+
+	return ret;
+}
+
+/**
+ * Set the enabled channels.
+ * @param dev - The device structure.
+ * @param channel_enable - The array contains channels enabled.
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+int32_t ad5770r_set_channel_en(struct ad5770r_dev *dev,
+			       const struct ad5770r_channel_switches *channel_enable)
+{
+	int32_t ret;
+
+	if (!dev || !channel_enable)
+		return FAILURE;
+
+	ret = ad5770r_spi_reg_write(dev,
+				    AD5770R_CH_ENABLE,
+				    AD5770R_CH_ENABLE_SET(channel_enable->en0, AD5770R_CH0) |
+				    AD5770R_CH_ENABLE_SET(channel_enable->en1, AD5770R_CH1) |
+				    AD5770R_CH_ENABLE_SET(channel_enable->en2, AD5770R_CH2) |
+				    AD5770R_CH_ENABLE_SET(channel_enable->en3, AD5770R_CH3) |
+				    AD5770R_CH_ENABLE_SET(channel_enable->en4, AD5770R_CH4) |
+				    AD5770R_CH_ENABLE_SET(channel_enable->en5, AD5770R_CH5));
+
+	if (ret)
+		return ret;
+
+	dev->channel_enable = *channel_enable;
+
+	return ret;
+}
+
+/**
+ * Get status value.
+ * @param dev - The device structure.
+ * @param mon_setup - The structure that contains the monitor setup
+ *		values.
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+int32_t ad5770r_get_status(struct ad5770r_dev *dev,
+			   uint8_t *status)
+{
+
+	if (!dev)
```

Comment:
nitpick: extra line before this check;
also, `ad5770r_spi_reg_read()` handles NULL, so you could remove this check